Framework/Crystal/inc/MantidCrystal/PeakStatisticsTools.h

```diff
@@ -0,0 +1,144 @@
+ * specified d-limits, given a certain unit cell, lattice centering
+ * and point group. The cost of this computation depends directly
+ * on the size of the unit cell (larger cells result in more
+ * reflections) and to some extent also on the symmetry (higher symmetry
+ * results in more matrix operations).
+ *
+ * After adding observations using addObservations, various reflection-
+ * counts can be obtained, for example to calculate redundancy or
+ * completeness of the observations.
+ *
+ */
+class DLLExport UniqueReflectionCollection {
+public:
+  UniqueReflectionCollection(
+      const Geometry::UnitCell &cell, const std::pair<double, double> &dLimits,
+      const Geometry::PointGroup_sptr &pointGroup,
+      const Geometry::ReflectionCondition_sptr &centering);
+
+  ~UniqueReflectionCollection() = default;
+
+  void addObservations(const std::vector<DataObjects::Peak> &peaks);
+  UniqueReflection getReflection(const Kernel::V3D &hkl) const;
+
+  size_t getUniqueReflectionCount() const;
+  size_t getObservedUniqueReflectionCount(size_t moreThan = 0) const;
+  std::vector<Kernel::V3D> getUnobservedUniqueReflections() const;
+
+  size_t getObservedReflectionCount() const;
+
+  const std::map<Kernel::V3D, UniqueReflection> &getReflections() const;
+
+protected:
+  /// Alternative constructor for testing purposes, no validation is performed.
+  UniqueReflectionCollection(
+      const std::map<Kernel::V3D, UniqueReflection> &reflections,
+      const Geometry::PointGroup_sptr &pointGroup)
+      : m_reflections(reflections), m_pointgroup(pointGroup) {}
+
+private:
+  std::map<Kernel::V3D, UniqueReflection> m_reflections;
+  Geometry::PointGroup_sptr m_pointgroup;
+};
+
+/**
+ * \class PeaksStatistics
+ *
+ * The PeaksStatistics class is a small helper class that is used
+ * in SortHKL. It takes a UniqueReflectionCollection and calculates
+ * a few data set quality indicators such as Rmerge and Rpim.
+ *
+ * Do not rely on this class to exist forever, parts of it may change
+ * or the entire class may disappear over time.
+ */
+class DLLExport PeaksStatistics {
+public:
+  explicit PeaksStatistics(const UniqueReflectionCollection &reflections)
+      : m_measuredReflections(0), m_uniqueReflections(0), m_completeness(0.0),
+        m_redundancy(0.0), m_rMerge(0.0), m_rPim(0.0), m_meanIOverSigma(0.0),
+        m_dspacingMin(0.0), m_dspacingMax(0.0), m_chiSquared(0.0), m_peaks() {
+    m_peaks.reserve(reflections.getObservedReflectionCount());
+    calculatePeaksStatistics(reflections.getReflections());
+  }
+
+  int m_measuredReflections;
+  int m_uniqueReflections;
+  double m_completeness;
+  double m_redundancy;
+  double m_rMerge;
+  double m_rPim;
+  double m_meanIOverSigma;
+  double m_dspacingMin;
+  double m_dspacingMax;
+  double m_chiSquared;
+
+  std::vector<DataObjects::Peak> m_peaks;
+
+private:
+  void calculatePeaksStatistics(
+      std::map<Kernel::V3D, UniqueReflection> uniqueReflections);
```

Comment:
`UniqueReflection` could get a `remove_outliers` parameter in the constructor. Then a new `UniqueReflection` would be constructed with the parameter set. There'd still be a copy involved, but it would be slightly less bad I suppose. Do you have another idea?